packet/flatcar-linux/kubernetes/workers/cl/worker.yaml.tmpl

```diff
@@ -131,30 +131,96 @@ storage:
+            local disk_type="$2"
+            # RAID device path, starting with /dev/
+            local device_path="$3"
+            local setup_fs_on_raid="$4"
 
-            cat /etc/mdadm.conf | while read line; do
-              # Make mount persistent across reboots.
-              local devname=$(echo "$${line}" | awk '{print $2}')
+            # Select disks for RAID sorted by size, filter by disk type and
+            # ignore the disks which are mounted (e.g., where Linux is installed)
+            local disks=$(lsblk -lnpd -x size -o path,rota -I "$${major_numbers}" \
+            | (
+            while IFS= read -r line; do
+              local drive=$(echo "$line" | awk '{print $1}')
+              local rota=$(echo "$line" | awk '{print $2}')
+              if [ $$disk_type != -1 ] && [ $$disk_type != $$rota ]; then
+                continue
+              fi
+              local mountpoints=$(lsblk -ln -o mountpoint "$drive")
+              if [[ -z "$mountpoints" ]]; then
+                echo "$line"
+              fi
+            done) | awk '{x=$1 " " x;} END{print x}')
 
-              # Mount the RAID device only if there is a filesystem on the device
-              local fs=$(blkid "$${devname}")
-              [ -z "$${fs}" ] && continue
+            local count=$(echo "$$disks" | wc -w)
 
-              local mount_point=$(basename "$${devname}")
-              # Create the mount point before mounting
-              mkdir "/mnt/$${mount_point}"
-              mount "$${devname}" "/mnt/$${mount_point}"
-              echo "$${devname} /mnt ext4 defaults,nofail,discard 0 0" | tee -a /etc/fstab
+            # Exit if we don't have any disks to create an array
+            [ $$count -lt 1 ] && return 0
+
+            # Create, format and mount array.
+            local extra_opts=""
+            if [ $$count -lt 2 ]; then
+              # Force array creation even with one disk
+              extra_opts="--force"
+            fi
+
+            # If the device_path is /dev/md/node-local-hdd-storage then the
+            # array name would be node-local-hdd-storage
+            array_name=$(basename "$${device_path}")
+
+            mdadm --create "$${device_path}" \
+              --homehost=any \
+              $$extra_opts \
+              --verbose \
+              --name="$${array_name}" \
+              --level=0 \
+              --raid-devices="$${count}" \
+              $${disks}
+
+            cat /proc/mdstat
+            # Wait for udev to create the symlinks
+            while [ ! -L "$${device_path}" ]; do
+              echo "Waiting for $${device_path}"
+              sleep 1
             done
+
+            mdadm --detail --scan | grep "ARRAY $${device_path} " | tee -a /etc/mdadm.conf
+
+            if [ "$${setup_fs_on_raid}" = true ]; then
+              mkfs.ext4 "$${device_path}"
+              mkdir "/mnt/$${array_name}"
+              mount "$${device_path}" "/mnt/$${array_name}"
```

Comment:
This changes the current behaviour of mounting it all on `/mnt/`. We are using this fact in some clusters, but not sure if it's worth having the special case here or trying to adapt clusters to this...

Will have to investigate a little bit :)

Reply:
That's already done on master, there is no change here.

Reply:
Hmm, not really, I think. Because on master there is code duplication and you have this function:

https://github.com/kinvolk/lokomotive-kubernetes/blob/master/packet/flatcar-linux/kubernetes/workers/cl/install.yaml.tmpl#L63-L89

And then this:

https://github.com/kinvolk/lokomotive-kubernetes/blob/master/packet/flatcar-linux/kubernetes/workers/cl/worker.yaml.tmpl#L128-L155

So, it is mounted to `/mnt` when you use `setup_raid="true"` on terraform.

And if for some reason this was recently broken, then that change did a backwards incompatible change on the raid mount point.

Reply:
That's the line that mounts it: https://github.com/kinvolk/lokomotive-kubernetes/blob/master/packet/flatcar-linux/kubernetes/workers/cl/worker.yaml.tmpl#L152
And it's not done on `/mnt` anymore.

Reply:
Ouch, so we already broke it. Then, go ahead. If we need to fix it, we can do it in another PR (I think we maybe can avoid to fix this and just have it this way :))